backend/api/src/realms.rs

```diff
@@ -3,49 +3,49 @@ use futures::{stream::TryStreamExt, TryStream};
 use juniper::graphql_object;
 use std::collections::HashMap;
 
-use super::Context;
+use crate::{Context, Id, Key};
 
 
-pub(super) type Id = i32;
+pub const KIND_PREFIX: &[u8; 2] = b"re";
```

Comment:
You make this `pub`. I used `pub(super)` (I think) consistently below. Should this also be `pub(super)`? Or should **everything** be `pub`?

Obviously you know better what's more idiomatic. My thinking was: The stricter the better for now; you can always make it less strict later, while the other direction is harder.

Reply:
I added a commit that changes most visibilities to `pub(crate)`. The most important visibilities are certainly module-private and public. That is followed by `pub(crate)`. Making something public for only a very specific module or so is very rare and usually not worth it.